source/Nevermore.Contracts/IId.cs

```diff
@@ -4,4 +4,18 @@ public interface IId
     {
         string Id { get; }
     }
-}
+
+    public interface IId<out TId> where TId : IIdWrapper
+    {
+        TId Id { get; }
+    }
+
+    /// <summary>
+    /// Implementations must have a constructor that takes a single string value
+    /// <see cref="Nevermore.Mapping.AmazingConverter"/>
+    /// </summary>
+    public interface IIdWrapper
```

Comment:
Should this really be `IStringWrapper` (or even just `ITypedString`), so that it can be used just as easily with other properties as with IDs? We might be missing some test cases for documents with typed strings on non-ID properties.

Reply:
Does Nevermore need to know about typed strings in other properties? that's probably not Nevermore's concern?

Reply:
Actually we missed a crucial case: allowing custom types in columns (if a property maps to a column), it blows up currently: https://github.com/OctopusDeploy/Nevermore/blob/f9fbe8a87bcc79382b5154f50727b8fdeb402169/source/Nevermore/Mapping/DatabaseTypeConverter.cs#L9

Reply:
Ah you can fix it by calling
`Column(m => m.NotProductId, c => c.DbType = DbType.String);`